source/Irrlicht/CIrrDeviceSDL.cpp

```diff
@@ -908,6 +933,83 @@ void CIrrDeviceSDL::setWindowCaption(const wchar_t* text)
 }
 
 
+//! Sets the window icon.
+bool CIrrDeviceSDL::setWindowIcon(const std::string &path)
+{
+	if (!Window)
+		return false;
+
+	video::IImageLoader *image_loader = nullptr;
+	u32 cnt = VideoDriver->getImageLoaderCount();
+	for (u32 i = 0; i < cnt; i++) {
+		if (VideoDriver->getImageLoader(i)->isALoadableFileExtension(
+					path.c_str())) {
+			image_loader = VideoDriver->getImageLoader(i);
+			break;
+		}
+	}
+
+	if (!image_loader) {
+		os::Printer::log(("Could not find image loader for file '" + path + "'").c_str(), ELL_WARNING);
+		return false;
+	}
+
+	io::IReadFile *icon_f = getFileSystem()->createAndOpenFile(path.c_str());
+
+	if (!icon_f) {
+		os::Printer::log(("Could not load icon file '" + path + "'").c_str(), ELL_WARNING);
+		return false;
+	}
+
+	video::IImage *img = image_loader->loadImage(icon_f);

```

Comment:
Why not `VideoDriver->createImageFromFile(path)`?

Reply:
I've adapted the code from the X11 thing we had in renderingengine.cpp. I didn't know this function exists.

Using `VideoDriver->createImageFromFile(path)` now, thanks!